### PR TITLE
Code Quality Reports CI Template Follow-Up

### PR DESCRIPTION
**Pull request by alzimmermsft**
# Description

This PR is a follow-up to #26140 which resolves issues found in the pipeline when ran. This is a follow-up as it wasn't possible to run the pipeline without committing the ci.yml file contained in the previous PR.

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#general-guidelines)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](../../CONTRIBUTING.md#building-and-unit-testing)
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

**Review by benbp — APPROVED** [review]

**Comment by alzimmermsft** [comment]
Failing pipelines aren't due to changes made in this PR, bypassing them.

File: eng/scripts/generate_from_source_pom.py [hunk 1]
```diff
@@ -118,7 +119,18 @@ def create_from_source_pom(project_list: str, set_pipeline_variable: str):
         fromSourcePom.write(pom_file_end)
 
     if set_pipeline_variable:
-        print('##vso[task.setvariable variable={};]{}'.format(set_pipeline_variable, json.dumps(modules)))
+        checkout_paths = list(set(sorted([p.directory_path for p in source_projects])))
```
Comment (benbp) [inline eng/scripts/generate_from_source_pom.py:122]:
```suggestion
        checkout_paths = sorted(list(set([p.directory_path for p in source_projects])))
```

File: eng/scripts/generate_from_source_pom.py [hunk 2]
```diff
@@ -101,14 +102,14 @@ def create_from_source_pom(project_list: str, set_pipeline_variable: str):
...
-    add_module_paths(modules, dependent_modules, projects)
-    add_module_paths(modules, dependency_modules, projects)
-
-    modules = sorted(modules)
+    # Finally map the project identifiers to projects.
+    add_source_projects(source_projects, project_list_identifiers, projects)
+    add_source_projects(source_projects, dependent_modules, projects)
+    add_source_projects(source_projects, dependency_modules, projects)
+    
+    modules = list(set(sorted([p.module_path for p in source_projects])))
```
Comment (benbp) [inline eng/scripts/generate_from_source_pom.py:112]: I think you sort/set/list this variable three times. Might be simpler to do it once?

**Review by benbp — APPROVED** [review]

**Pull request merged**